### PR TITLE
feat: `shield:setup` does Email setup

### PR DESCRIPTION
**Pull request by kenjis**
~~Needs #871~~

**Description**
- `shield:setup` does Email setup
- refactor

**Checklist:**
- [x] Securely signed commits
- [ ] Component(s) with PHPDoc blocks, only if necessary or adds value
- [ ] Unit testing, with >80% coverage
- [x] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

File: src/Commands/BaseCommand.php [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+     *
+     * @param string       $field      Output "field" question
+     * @param array|string $options    String to a default value, array to a list of options (the first option will be the default value)
+     * @param array|string $validation Validation rules
+     *
+     * @return string The user input
+     */
+    protected function prompt(string $field, $options = null, $validation = null): string
+    {
+        return self::$io->prompt($field, $options, $validation);
```
Comment (datamweb) [inline src/Commands/BaseCommand.php:33]:
```suggestion
        $this->ensureInputOutput();
        return self::$io->prompt($field, $options, $validation);
```

**Comment by datamweb** [comment]
Just note, resulte tests in windows:
```console
There were 4 failures:

1) Tests\Commands\UserTest::testList
Failed asserting that 'Id       User\r\n
1       user8 (user8@example.com)\r\n
2       user9 (user9@example.com)\r\n
' contains "Id  User
1       user8 (user8@example.com)
2       user9 (user9@example.com)
".

P:\MyGitHubWork\shield\tests\Commands\UserTest.php:464

2) Tests\Commands\UserTest::testListByEmail
Failed asserting that 'Id       User\r\n
2       user9 (user9@example.com)\r\n
' contains "Id  User
2       user9 (user9@example.com)
".

P:\MyGitHubWork\shield\tests\Commands\UserTest.php:490

3) Tests\Commands\SetupTest::testRunEmailConfigIsFine
Failed asserting that '  Created: vfs://root/Config/Auth.php\r\n
  Created: vfs://root/Config/AuthGroups.php\r\n
  Created: vfs://root/Config/AuthToken.php\r\n
  Updated: vfs://root/Controllers/BaseController.php\r\n
  Updated: vfs://root/Config/Routes.php\r\n
  Updated: We have updated file 'vfs://root/Config/Security.php' for security reasons.\r\n
  Email Setup: Everything is fine.\r\n
Running all new migrations...\r\n
Migrations complete.\r\n
\r\n
' contains "  Created: vfs://root/Config/Auth.php
  Created: vfs://root/Config/AuthGroups.php
  Created: vfs://root/Config/AuthToken.php
  Updated: vfs://root/Controllers/BaseController.php
  Updated: vfs://root/Config/Routes.php
  Updated: We have updated file 'vfs://root/Config/Security.php' for security reasons.".

P:\MyGitHubWork\shield\tests\Commands\SetupTest.php:111

4) Tests\Commands\SetupTest::testRun
Failed asserting that '  Created: vfs://root/Config/Auth.php\r\n
  Created: vfs://root/Config/AuthGroups.php\r\n
  Created: vfs://root/Config/AuthToken.php\r\n
  Updated: vfs://root/Controllers/BaseController.php\r\n
  Updated: vfs://root/Config/Routes.php\r\n
  Updated: We have updated file 'vfs://root/Config/Security.php' for security reasons.\r\n
  Updated: vfs://root/Config/Email.php\r\n
Running all new migrations...\r\n
Migrations complete.\r\n
\r\n
' contains "  Created: vfs://root/Config/Auth.php
  Created: vfs://root/Config/AuthGroups.php
  Created: vfs://root/Config/AuthToken.php
  Updated: vfs://root/Controllers/BaseController.php
  Updated: vfs://root/Config/Routes.php
  Updated: We have updated file 'vfs://root/Config/Security.php' for security reasons.
  Updated: vfs://root/Config/Email.php".

P:\MyGitHubWork\shield\tests\Commands\SetupTest.php:76

FAILURES!
Tests: 466, Assertions: 1029, Failures: 4.
```

File: src/Commands/BaseCommand.php (hunk 1 above)
Comment (kenjis) [reply src/Commands/BaseCommand.php:42]: Added it in the constructor.

**Review by datamweb — APPROVED** [review]
LGTM!
Thank you!

**Comment by kenjis** [comment]
@datamweb Thank you for review!

**Pull request merged**